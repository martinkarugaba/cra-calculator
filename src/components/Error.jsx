const Error = () => (
  <div className="h-[100vh] w-screen flex justify-center items-center">
    <h1 className="text-5xl font-semibold">There was an error!</h1>
  </div>
);
export default Error;
