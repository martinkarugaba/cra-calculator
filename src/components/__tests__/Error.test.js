import React from 'react';
import { render } from '@testing-library/react';
import Error from '../Error';

describe('Error component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
  });

  it('renders the error message correctly', () => {
    const { getByText } = render(<Error />);
    const errorMessage = getByText('There was an error!');
    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toHaveClass('text-5xl font-semibold');
  });
});
