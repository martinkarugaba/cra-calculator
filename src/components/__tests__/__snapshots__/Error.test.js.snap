// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component matches the snapshot 1`] = `
<div>
  <div
    class="h-[100vh] w-screen flex justify-center items-center"
  >
    <h1
      class="text-5xl font-semibold"
    >
      There was an error!
    </h1>
  </div>
</div>
`;
